lib/rust/types/src/unit2.rs

```diff
@@ -873,3 +875,77 @@ impl<'t> BytesCowOps<'t, ops::RangeFrom<Bytes>> for Cow<'t, str> {
         }
     }
 }
+
+
+
+// ================
+// === Fraction ===
+// ================
+
+define! {
+    /// A fraction of the leftover space, for example in the grid container. It is used in auto
+    /// layouts, similar to the CSS `fr` unit: https://www.w3.org/TR/css3-grid-layout/#fr-unit.
+    Fraction: f32 = 0.0
+}
+define_ops![
+    Fraction [+,-] Fraction = Fraction,
+    Fraction * f32 = Fraction,
+    f32 * Fraction = Fraction,
+];
+
+/// Constructors.
+#[allow(missing_docs)]
+pub trait FractionOps {
+    fn fr(&self) -> Fraction;
+}
+
+impl FractionOps for f32 {
+    fn fr(&self) -> Fraction {
+        Fraction::unchecked_from(*self)
+    }
+}
+
+impl FractionOps for i32 {
+    fn fr(&self) -> Fraction {
+        (*self as f32).fr()
+    }
+}
+
+impl From<f32> for Fraction {
+    fn from(t: f32) -> Self {
+        Fraction::unchecked_from(t)
+    }
+}
+
+
+// ===============
+// === Percent ===
+// ===============
+
+define! {
+    /// A percentage value.
+    Percent: f32 = 0.0
+}
+define_ops![
+    Percent [+,-] Percent = Percent,
+    Percent * f32 = Percent,
+    f32 * Percent = Percent,
+];
+
+/// Constructors.
+#[allow(missing_docs)]
+pub trait PercentOps {
+    fn pc(&self) -> Percent;
+}
+
+impl PercentOps for f32 {
+    fn pc(&self) -> Percent {
+        Percent::unchecked_from(*self)
+    }
+}
+
+impl PercentOps for i32 {
+    fn pc(&self) -> Percent {
+        (*self as f32).pc()
+    }
+}
```

Comment:
Actualy, I think Fraction and Percent should have defined Mul for every f32 unit.

Reply:
It's a lot of additional efford. I agree with that, but let's do it not as part of this PR, please.